Runtime/src/index.ts

```diff
@@ -383,12 +385,18 @@ export class SwiftRuntime {
                 payload2_ptr: pointer
             ) => {
                 const func = this.heap.referenceHeap(ref);
-                const result = Reflect.apply(
-                    func,
-                    undefined,
-                    decodeValues(argv, argc)
-                );
-                writeValue(result, kind_ptr, payload1_ptr, payload2_ptr);
+                let result: any;
+                try {
+                    result = Reflect.apply(
+                        func,
+                        undefined,
+                        decodeValues(argv, argc)
+                    );
+                } catch (error) {
+                    writeValue(error, kind_ptr, payload1_ptr, payload2_ptr, true);
+                    return;
+                }
+                writeValue(result, kind_ptr, payload1_ptr, payload2_ptr, false);
```

Comment:
Why can’t this be moved inside the `try`?

```js
writeValue(
  Reflect.apply(
    func,
    undefined,
    decodeValues(argv, argc)
  ),
  kind_ptr,
  payload1_ptr,
  payload2_ptr,
  false
);
```

Reply:
It's intentional. `writeValue` can throw exceptions but it's system errors and users don't want to catch those kinds of system errors.